programs/fund-sender/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+#![allow(clippy::result_large_err)]
+use crate::utils::errors::ErrorCode;
+use crate::utils::spend::*;
+use crate::utils::state::*;
+use anchor_lang::prelude::*;
+mod utils;
+
+// how to write it to leave it to be filled with new id for each deployed program for each climate project?
+declare_id!("rodTth5pXjkUfQpqMp7tEFdN1sdv2JwqhXg8RH9YrWD");
+
+#[program]
+pub mod fund_sender {
+    use super::*;
+
+    pub fn register_state(
+        ctx: Context<RegisterState>,
+        sunrise_state: Pubkey,
+        state_in: GenericStateInput,
+    ) -> Result<()> {
+        // register state account on chain, only need to ever be done once
+        let state = &mut ctx.accounts.state;
+        state.sunrise_state = sunrise_state;
+        state.update_authority = state_in.update_authority;
+        state.destination_seed = state_in.destination_seed;
+        state.destination_account = state_in.destination_account;
+        state.spend_threshold = state_in.spend_threshold;
+        state.output_yield_account_bump = *ctx.bumps.get("output_yield_account").unwrap();
+        state.total_spent = 0;
+
+        Ok(())
+    }
+
+    pub fn update_state(ctx: Context<UpdateState>, state_in: GenericStateInput) -> Result<()> {
+        // update state account parameters
+        let state = &mut ctx.accounts.state;
+        state.update_authority = state_in.update_authority;
+        state.destination_account = state_in.destination_account;
+        state.spend_threshold = state_in.spend_threshold;
+
+        Ok(())
+    }
+
+    pub fn send_fund<'info>(ctx: Context<'_, '_, '_, 'info, SendFund<'info>>) -> Result<()> {
+        // send yield to output_yield_accounts with specified proportions
+        let state = &mut ctx.accounts.state;
+        let output_yield_account = &mut ctx.accounts.output_yield_account;
+        let destination_account = &mut ctx.accounts.destination_account;
```

Comment:
We still need to check if this is the correct destination account with
`if destination_account.key() != state.destination_account {
            return Err(ErrorCode::IncorrectDestinationAccount.into());
        }`

Or we can do the check in the account struct.